src/Controller/Action/CleanWishlistAction.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusWishlistPlugin\Controller\Action;
+
+use BitBag\SyliusWishlistPlugin\Context\WishlistContextInterface;
+use BitBag\SyliusWishlistPlugin\Entity\WishlistInterface;
+use BitBag\SyliusWishlistPlugin\Repository\WishlistRepositoryInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+final class CleanWishlistAction
+{
+    private WishlistContextInterface $wishlistContext;
+
+    private WishlistRepositoryInterface $wishlistRepository;
+
+    private EntityManagerInterface $wishlistManager;
+
+    private FlashBagInterface $flashBag;
+
+    private TranslatorInterface $translator;
+
+    private UrlGeneratorInterface $urlGenerator;
+
+    public function __construct(
+        WishlistContextInterface $wishlistContext,
+        WishlistRepositoryInterface $wishlistRepository,
+        EntityManagerInterface $wishlistManager,
+        FlashBagInterface $flashBag,
+        TranslatorInterface $translator,
+        UrlGeneratorInterface $urlGenerator
+    ) {
+        $this->wishlistContext = $wishlistContext;
+        $this->wishlistRepository = $wishlistRepository;
+        $this->wishlistManager = $wishlistManager;
+        $this->urlGenerator = $urlGenerator;
+        $this->flashBag = $flashBag;
+        $this->translator = $translator;
+    }
+
+    public function __invoke(int $wishlistId, Request $request): Response
+    {
+        /** @var WishlistInterface $wishlist */
+        $wishlist = $this->wishlistRepository->find($wishlistId);
+
+        $wishlistProducts = $wishlist->getWishlistProducts();
```

Comment:
This can be definitely an entity method `$wishlist->clear()` and this loop logic to be done inside.

Reply:
Cleaning wishlist will be correct in future, on a separate branch